Subject: [patch] xf86-video-fbdev: add option to leave timings alone
From: Sascha Hauer <s.hauer@pengutronix.de>

Xorg tries to set the timing values in the framebuffer device. Add an
option to tell X to leave the timing values untouched.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>

---
FIXME rsc: what's the purpose of this patch?

Index: xf86-video-fbdev-0.3.1/src/fbdev.c
===================================================================
--- xf86-video-fbdev-0.3.1.orig/src/fbdev.c	2006-12-29 12:39:35.000000000 +0100
+++ xf86-video-fbdev-0.3.1/src/fbdev.c	2007-10-18 18:47:00.000000000 +0200
@@ -104,6 +104,7 @@
 	OPTION_SHADOW_FB,
 	OPTION_ROTATE,
 	OPTION_FBDEV,
+	OPTION_DONT_SET_TIMING,
 	OPTION_DEBUG
 } FBDevOpts;
 
@@ -111,6 +112,7 @@
 	{ OPTION_SHADOW_FB,	"ShadowFB",	OPTV_BOOLEAN,	{0},	FALSE },
 	{ OPTION_ROTATE,	"Rotate",	OPTV_STRING,	{0},	FALSE },
 	{ OPTION_FBDEV,		"fbdev",	OPTV_STRING,	{0},	FALSE },
+	{ OPTION_DONT_SET_TIMING, "DontSetTiming", OPTV_BOOLEAN, {0},	FALSE },
 	{ OPTION_DEBUG,		"debug",	OPTV_BOOLEAN,	{0},	FALSE },
 	{ -1,			NULL,		OPTV_NONE,	{0},	FALSE }
 };
@@ -503,9 +505,11 @@
 	}
 
 	/* select video modes */
-
-	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against framebuffer device...\n");
-	fbdevHWSetVideoModes(pScrn);
+	if( xf86ReturnOptValBool(fPtr->Options, OPTION_DONT_SET_TIMING, FALSE) == FALSE ) {
+		xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against framebuffer device...\n");
+		fbdevHWSetVideoModes(pScrn);
+	} else
+		xf86DrvMsg(pScrn->scrnIndex, X_INFO, "ignoring fb timing settings.");
 
 	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against monitor...\n");
 	{
